unified-runtime/source/adapters/offload/memory.hpp

```diff
@@ -22,6 +22,23 @@ struct BufferMem {
     AllocHostPtr,
   };
 
+  struct BufferMap {
+    size_t MapSize;
+    size_t MapOffset;
+    ur_map_flags_t MapFlags;
+    // Allocated host memory used exclusively for this map.
+    std::shared_ptr<unsigned char[]> MapMem;
```

Comment:
Why a shared_ptr rather than a unique_ptr?

Reply:
Honestly not sure, this was a copy+paste from other adapters. I've changed it to unique_ptr now, that makes more sense